src/analysis/vector/geometry_checker/qgsgeometrygapcheck.cpp

```diff
@@ -248,13 +252,51 @@ void QgsGeometryGapCheck::fixError( const QMap<QString, QgsFeaturePool *> &featu
         }
         break;
       }
+
+      case CreateNewFeature:
+      {
+        QgsGeometryGapCheckError *gapCheckError = dynamic_cast<QgsGeometryGapCheckError *>( error );
+        QgsProject *project = QgsProject::instance();
```

Comment:
Does this always get run from the main thread?

Reply:
Yes

https://github.com/qgis/QGIS/blob/master/src/analysis/vector/geometry_checker/qgsgeometrycheck.h#L306-L312